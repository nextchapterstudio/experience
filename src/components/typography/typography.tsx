import React, {  ElementType, FC  } from 'react'
import classnames from 'classnames'
// import { BaseElement } from 'components/base-element'


export const Typography: FC<TypographyProps> = ({
    as: Element = 'span',
    className,
    is,
    isSpaced,
    children,
    ...props
  }) => {
    const classes = {
      [`is-${is}`]: is,
      'is-spaced': isSpaced,
    }
  
    return (
      <Element className={classnames('typography', className, classes)} {...props}>
        {children}
      </Element>
    )
  }
  export interface TypographyProps {
    as?: ElementType;
    children?: React.ReactNode
    className?: string
    isSpaced?: boolean
    is?: '1' | '2' | '3' | '4' | '5' | '6'| false
    status?: 'inactive' | 'inProgress' | 'success' | 'failed'
    value?: string | number | Error
  }

  
  Typography.defaultProps = {
    as: 'span',
    is: false,
    isSpaced: false,
  }
  