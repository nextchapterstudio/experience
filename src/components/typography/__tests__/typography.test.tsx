import React from 'react'
import { render, cleanup, fireEvent, screen } from '@testing-library/react'
import { Typography } from '../typography'

describe('The Typography Component', () => {
  afterEach(cleanup)
  it('should render a div with the class basic', () => {
    const { container } = render(<Typography className='basic'>test</Typography>)

    expect(container).toMatchSnapshot()
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('basic')
    expect(container.firstChild).toHaveClass('typography')
  })

  it('should render a span with the class basic', () => {
    const { container } = render(<Typography as="span" className='basic' >test</Typography>)

    expect(container).toMatchSnapshot()
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('basic')
    expect(container.firstChild).toHaveClass('typography')
  })
  it('should render a div with the html attribute', () => {
    const { container } = render(<Typography className='basic' aria-hidden>test</Typography>)

    expect(container).toMatchSnapshot()
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveAttribute('aria-hidden')
    expect(container.firstChild).toHaveClass('typography')
  })
  it('should render as a p tag', () => {
    const { container } = render(<Typography className='basic' as="p">test</Typography>)
    expect(container).toMatchSnapshot()
    expect(container).toContainHTML('<p')
    expect(container.firstChild).toHaveClass('typography')
  })
})
