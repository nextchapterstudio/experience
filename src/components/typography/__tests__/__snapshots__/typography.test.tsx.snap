// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Typography Component should render a div with the class basic 1`] = `
<div>
  <span
    class="typography basic"
  >
    test
  </span>
</div>
`;

exports[`The Typography Component should render a div with the class basic 2`] = `
<span
  class="typography basic"
>
  test
</span>
`;

exports[`The Typography Component should render a div with the html attribute 1`] = `
<div>
  <span
    aria-hidden="true"
    class="typography basic"
  >
    test
  </span>
</div>
`;

exports[`The Typography Component should render a div with the html attribute 2`] = `
<span
  aria-hidden="true"
  class="typography basic"
>
  test
</span>
`;

exports[`The Typography Component should render a span with the class basic 1`] = `
<div>
  <span
    class="typography basic"
  >
    test
  </span>
</div>
`;

exports[`The Typography Component should render a span with the class basic 2`] = `
<span
  class="typography basic"
>
  test
</span>
`;

exports[`The Typography Component should render as a p tag 1`] = `
<div>
  <p
    class="typography basic"
  >
    test
  </p>
</div>
`;
